commands/volumes/volume-status.go

```diff
@@ -48,12 +50,34 @@ func checkStatus(ctx transaction.TxnCtx) error {
 			continue
 		}
 
-		// TODO: Check actual brick status when we get them running.
-		fakeStatus := &brick.Brickstatus{
-			Online: false,
-			Pid:    1234,
+		port := pmap.RegistrySearch(binfo.Path, pmap.GfPmapPortBrickserver)
+
+		brickDaemon, err := brick.NewGlusterfsd(binfo)
+		if err != nil {
+			return err
+		}
+
+		online := false
+
+		pid, err := daemon.ReadPidFromFile(brickDaemon.PidFile())
+		if err == nil {
+			// Check if process is running
+			_, err := daemon.GetProcess(pid)
+			if err == nil {
+				online = true
+			}
+			// Set port to 0 in case the brick process is not running
+			// but pmap.RegistrySearch() returns stale port information
+			port = 0
```

Comment:
This will unconditionally set the port to 0. Shouldn't you be doing the following (example below) after the entire if block:

```go
if !online {
    port = 0
}
```

OR just move line 53 into the place where you set `online = true` 

```go

var port int

pid, err := daemon.ReadPidFromFile(brickDaemon.PidFile())
if err == nil {
		_, err := daemon.GetProcess(pid)
		if err == nil {
			online = true
                        port = pmap.RegistrySearch(binfo.Path, pmap.GfPmapPortBrickserver)
		}
}

````